### PR TITLE
#96 - Uploading smaller content using put request in S3Storage

### PR DESCRIPTION
**Pull request by olegmoz**
Part of issue #87 
`S3Storage.save()` uses put request when content is small, so the operation is done only single call and so much more efficient

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 Please assign me

**Comment by 0crat** [comment]
This pull request [#118](https://github.com/artipie/asto/issues/118) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/edc8faaa-2b0d-42c5-acf3-9393563d7d28); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/asto/s3/S3Storage.java [hunk 1]
```diff
@@ -122,34 +130,57 @@ public S3Storage(final S3AsyncClient client, final String bucket) {
...
-                        );
+        final CompletableFuture<Void> future;
+        final Optional<Long> size = content.size();
+        if (size.isPresent() && size.get() < S3Storage.MIN_MULTIPART) {
+            future = this.client.putObject(
+                PutObjectRequest.builder()
+                    .bucket(this.bucket)
+                    .key(key.string())
+                    .build(),
+                new AsyncRequestBody() {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/asto/s3/S3Storage.java]: @olegmoz let's extract this to a private static class to make it more readable. The reader don't really care about the inner structure of async request so let's hide it. We can have something like `ContentBody`.

Comment (olegmoz) [reply src/main/java/com/artipie/asto/s3/S3Storage.java]: @Vatavuk created a private class for that, think it's event better. Please review

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Pull request merged**